packages/interactivity/src/hooks.tsx

```diff
@@ -257,12 +257,19 @@ export const getEvaluate: GetEvaluate =
 			}
 			// Reset scope before return and wrap the function so it will still run within the correct scope.
 			resetScope();
-			return ( ...functionArgs: any[] ) => {
+			const wrappedFunction: SyncAwareFunction = (
+				...functionArgs: any[]
+			) => {
```

Comment:
Small nit: I'm slightly wary of any side effects from typing this function unconditionally, even if the function isn't supposed to be `sync`.

Maybe safer to only do this `if ( value.sync )`?

I think you could assign it to an untyped variable here, and later reassign it, but only `if ( value.sync )`. This could be in the same clause where you actually set `wrappedFunction.sync = true`.